tests/testthat/test-memoise.R

```diff
@@ -326,6 +326,20 @@ test_that("omit_args respected", {
   res2 <- mem_rnorm(10, mean = +100)
 
   expect_true(identical(res1, res2))
+
+  # Also works for default args
+  f <- function(n, x = rnorm(1)) {
+    rnorm(n)
+  }
+  mem_f <- memoise(f)
+  expect_false(identical(res1, res2))
+  res1 <- mem_f(1)
+  res2 <- mem_f(1)
```

Comment:
The `expect_false` should come after assigning the values.

Reply:
Now that I think about it some more, can you change these tests to be something like this? The reason I ask for all these cases is because it helps make clear what the differences should be in different cases when `omit_args` is used.

```R
f <- function(n, x = rnorm(1)) {
  rnorm(n)
}

mem_f <- memoise::memoise(f)
expect_false(identical(
  mem_f(1),
  mem_f(1)
))
expect_identical(
  mem_f(1, 2),
  mem_f(1, 2)
)
expect_false(identical(
  mem_f(1, 3),
  mem_f(1, 4)
))


mem_f2 <- memoise::memoise(f, omit_args = "x")
expect_identical(
  mem_f2(1),
  mem_f2(1)
)
expect_identical(
  mem_f2(1, 2),
  mem_f2(1, 2)
)
expect_identical(
  mem_f2(1, 3),
  mem_f2(1, 4)
)
```

Reply:
I went for a completely different test logic which should hopefully be more clear overall.